driver-core/src/main/java/com/datastax/driver/core/QueryLogger.java

```diff
@@ -0,0 +1,544 @@
+                if (statement instanceof BoundStatement) {
+                    appendParameters((BoundStatement)statement, params, maxLoggedParameters);
+                } else if (statement instanceof BatchStatement) {
+                    BatchStatement batchStatement = (BatchStatement)statement;
+                    int remaining = maxLoggedParameters;
+                    for (Statement inner : batchStatement.getStatements()) {
+                        if (inner instanceof BoundStatement) {
+                            remaining = appendParameters((BoundStatement)inner, params, remaining);
+                        }
+                    }
+                }
+                if (params.length() > 0)
+                    params.append("]");
+                logger.trace(message + params, exception);
+            } else {
+                logger.debug(message, exception);
+            }
+        }
+    }
+
+    private String statementAsString(Statement statement) {
+        StringBuilder sb = new StringBuilder();
+        append(statement, sb, maxQueryStringLength);
+        return sb.toString();
+    }
+
+    private int appendParameters(BoundStatement statement, StringBuilder buffer, int remaining) {
+        if (remaining == 0)
+            return 0;
+        ColumnDefinitions metadata = statement.preparedStatement().getVariables();
+        int numberOfParameters = metadata.size();
+        if (numberOfParameters > 0) {
+            List<ColumnDefinitions.Definition> definitions = metadata.asList();
+            int numberOfLoggedParameters;
+            if (remaining == -1) {
+                numberOfLoggedParameters = numberOfParameters;
+            } else {
+                numberOfLoggedParameters = remaining > numberOfParameters ? numberOfParameters : remaining;
+                remaining -= numberOfLoggedParameters;
+            }
+            for (int i = 0; i < numberOfLoggedParameters; i++) {
+                if (buffer.length() == 0)
+                    buffer.append(" [");
+                else
+                    buffer.append(", ");
+                buffer.append(String.format("%s:%s", metadata.getName(i), parameterValueAsString(definitions.get(i), statement.values[i])));
+            }
+            if (numberOfLoggedParameters < numberOfParameters) {
+                buffer.append(FURTHER_PARAMS_OMITTED);
+            }
+        }
+        return remaining;
+    }
+
+    private String parameterValueAsString(ColumnDefinitions.Definition definition, ByteBuffer raw) {
+        String valueStr;
+        if (raw == null || raw.remaining() == 0) {
+            valueStr = "NULL";
+        } else {
+            DataType type = definition.getType();
+            int maxParameterValueLength = this.maxParameterValueLength;
+            if (type.equals(DataType.blob()) && maxParameterValueLength != -1) {
+                // prevent large blobs from being converted to strings
```

Comment:
As requested, we special-case blob types and try to avoid converting large buffers into strings.